app/src/main/java/org/wikipedia/page/PageToolbarHideHandler.java

```diff
@@ -56,12 +56,13 @@
      */
     void setFadeEnabled(boolean enabled) {
         fadeEnabled = enabled;
-        update();
+
+        // to avoid the toolbar applies the opacity before image loaded
+        pageFragment.getWebView().postDelayed(this::update, DELAY_UPDATE_MS);
     }
 
     @Override
     protected void onScrolled(int oldScrollY, int scrollY) {
-        tabsButton.setTabCount(WikipediaApp.getInstance().getTabCount());
 
         int opacity = fadeEnabled && scrollY < (pageFragment.getHeaderView().getHeight() - toolbarHeight) ? 0 : FULL_OPACITY;
```

Comment:
Instead of using `getHeaderView().getHeight()`, can we just use `DimenUtil.leadImageHeightForDevice()`? I believe that should fix it.

Reply:
Thanks, you're right!